File: queue/simpleQueue.java
//implementing the structure of queue

import java.util.*;

class Queue{
    int[] queue;
    int front, rear;
    
    Queue(int size){
        queue=new int[size];
        front=-1;
        rear=-1;
    }
    boolean isEmpty(){
        if (front==-1){
            return true;
        }
        else{
            return false;
        }
    }
    boolean isFull(){
        if (front==0 && rear==queue.length){
            return true;
        }
        else{
            return false;
        }
    }
    void enqueue(int element){
        if (isFull()){
            System.out.println("Queue is full");
        }
        else{
            if(isEmpty()){
                front=0;
            }
            queue[++rear]=element;
        }
    }
    void dequeue(){
        if (isEmpty()){
            System.out.println("Queue is empty");
        }
        else{
            System.out.println("Removed : "+queue[front]);
            if (front==rear){
                front=-1;
                rear=-1;
            }
            else{
                front++;
            }
        }
    }
    void display(){
        for(int i=front;i<=rear;i++){
            System.out.print(queue[i]+" ");
        }
        System.out.println();
    }
}


public class Main
{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int s = sc.nextInt();
		Queue q = new Queue(s);
		q.enqueue(10);
		q.enqueue(20);
		System.out.println(q.isEmpty());
		q.enqueue(30);
		q.display();
		q.dequeue();
		q.display();
	}
}

File: queue/simpleQueue.c
#include <stdio.h>
#include <stdlib.h>

struct Queue{
    int* arr;
    int front, rear, capacity;
};

struct Queue* createQueue(int size){
    struct Queue* queue = (struct Queue*)(malloc(sizeof(struct Queue)));
    queue->front=-1;
    queue->rear=-1;
    queue->capacity=size;
    queue->arr=(int*)(malloc(sizeof(int)*size));
    return queue;
}

int isEmpty(struct Queue* queue){
    return queue->front==-1 && queue->rear==-1;
}

int isFull(struct Queue* queue) {
    return queue->rear == queue->capacity - 1;
}


void enqueue(struct Queue* queue, int element){
    if(isFull(queue)){
        printf("Stack is full\n");
    }
    else{
        if(queue->front==-1){
            queue->front=0;
        }
        queue->arr[++(queue->rear)]=element;
    }
}

void dequeue(struct Queue* queue){
    if(isEmpty(queue)){
        printf("Stack is empty\n");
    }
    else{
        printf("Removed : %d\n",queue->arr[queue->front]);
        if(queue->front==queue->rear){
            queue->front=-1;
            queue->rear=-1;
        }
        else{
            queue->front++;
        }
    }
}

void display(struct Queue* queue){
    for(int i=queue->front;i<=queue->rear;i++){
        printf("%d ",queue->arr[i]);
    }
    printf("\n");
}

int main()
{
    int size;
    scanf("%d",&size);
    struct Queue* q = createQueue(size);
    enqueue(q,10);
    enqueue(q,20);
    enqueue(q,30);
    display(q);
    dequeue(q);
    display(q);
    enqueue(q,40);
    enqueue(q,50);
    enqueue(q,60);
    display(q);
    enqueue(q,70);
    dequeue(q);
    dequeue(q);
    dequeue(q);
    dequeue(q);
    dequeue(q);
    printf("%d %d",q->front,q->rear);
    return 0;
}
